analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -19,20 +19,265 @@
+            if (!license.isEmpty()) {
+                result.add(license)
+            }
+
+            return result.toSortedSet()
+        }
+
+        private fun extractPackage(node: JsonNode ): Package {
+            val vcsInfo = extractVcsInfo(node)
+            return Package(
+                    id = extractPackageId(node),
+                    declaredLicenses = extractDeclaredLicenses(node),
+                    description = node["pkgMeta"]["description"].textValueOrEmpty(),
+                    homepageUrl = node["pkgMeta"]["homepage"].textValueOrEmpty(),
+                    binaryArtifact = RemoteArtifact.EMPTY,
+                    sourceArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+                    vcs = vcsInfo,
+                    vcsProcessed = vcsInfo.normalize()
+            )
+        }
+
+        private fun extractPackages(node: JsonNode): Map<String, Package> {
+            val result = mutableMapOf<String, Package>()
+
+            val stack = Stack<JsonNode>()
+            stack.push(node)
+
+            while (!stack.empty()) {
+                val currentNode = stack.pop()
+                val pkg = extractPackage(currentNode)
+                result.put("${pkg.id.name}:${pkg.id.version}", pkg)
+
+                val children = currentNode["dependencies"].fields().asSequence().map { it.value }
+                stack.addAll(children)
+            }
+
+            return result
+        }
+
+        private fun extractDependencyTree(node: JsonNode, scopeName: String): SortedSet<PackageReference> {
+            val result = mutableSetOf<PackageReference>()
+
+            node["pkgMeta"][scopeName].fieldNamesOrEmpty().forEach {
+                val childNode = node["dependencies"][it]
+                val childScope = SCOPE_NAME_DEPENDENCIES
+                val childDependencies = extractDependencyTree(childNode, childScope)
+                val packageRefence = PackageReference(
+                        id = extractPackageId(childNode),
+                        dependencies = childDependencies
+                )
+                result.add(packageRefence)
+            }
+
+            return result.toSortedSet()
+        }
+    }
+
+    private val stashedDirectories = HashMap<File, File>()
+
+    override fun toString(): String {
+        return PROVIDER_NAME
+    }
+
+    override fun command(workingDir: File?) = if (OS.isWindows) "bower.cmd" else "bower"
+
+    override fun prepareResolution(definitionFiles: List<File>): List<File> {
+        jsonMapper.enable(SerializationFeature.INDENT_OUTPUT)
+        checkVersion(
+                Requirement.buildNPM(REQUIRED_BOWER_VERSION),
+                ignoreActualVersion = analyzerConfig.ignoreToolVersions
+        )
+        return definitionFiles
+    }
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        log.info( "Resolving dependencies for: $definitionFile" )
```

Comment:
Use this syntax to avoid building the String if `info` is not loggable:

```kotlin
log.info { "..." }
```

Reply:
Done.